postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py

```diff
@@ -243,7 +244,15 @@ def ApproximateEventDurationAndHypocenter(faultxdmf, faultxyz):
         ASl = LoadSingleComponentParallel("ASl", surfaceData=False)
     ASR = np.abs(np.gradient(ASl, dtFault, axis=1))
     slippingElement = np.where(np.amax(ASR, axis=0) > slipRateThreshold, 1, 0)
-    duration = slippingElement[np.argmax(slippingElement):-np.argmax(np.flip(slippingElement))].size * dtFault
+    if np.all(slippingElement == 1):
+        print("Warning: at least one element is constantly slipping")
+        print("eventDuration is set to the entire length of the time series")
+        duration = (slippingElement.size - 1) * dtFault
+    elif np.argmax(np.flip(slippingElement)) == 0:
+        print("Warning: at least one element is still slipping at the last time step")
+        duration = slippingElement[np.argmax(slippingElement):-1].size * dtFault
+    else:
+        duration = slippingElement[np.argmax(slippingElement):-np.argmax(np.flip(slippingElement))].size * dtFault
```

Comment:
why is this formula not valid for all cases?

Reply:
When there is still an element slipping at the last time step `-np.argmax(np.flip(slippingElement))` is zero, which means duration is also zero

Reply:
What about np.trim_zeros(slippingElement).size?

Reply:
Good idea, I changed it to np.trim_zeros